### PR TITLE
Integrate liquid and hybrid motors

### PR DESCRIPTION
**Pull request by GabrielBarberini**
# Liquid and hybrid motors integration

## 📝 Briefing 🎯

Currently [api.rocketpy.org](api.rocketpy.org) supports only solid motors, this commit is a design upgrade that enables solid, liquid, and hybrid motors to be used through the API. Additional enhancements in the design were also applied to serve as a scaffold for the next improvements such as allowing custom engine file input, selectable off-shelf engine files, etc.

## 🕵️ Review focus 🔎

I would like to kindly ask your help to check for performance improvements and flag any anti-patterns you find since In order to solve some of the integration issues I had to make some not-so-elegant work-around.

## 🔬 Tests 🧪

The Postman collection is yet to be updated with additional functional tests to cover this change. I've also tested every route for hybrid, solid, and liquid with the preset values and fixed all critical issues/bugs along the way. Still, I would like you to give it a try yourself to help me test a few edge cases, probably there are some covert bugs lying around.

## Additional info
- implements motor tanks
- implements guard on motor to cover hybrid and liquid inputs
- improves motors model readability
- improves motors model readability
- improves models visibility
- implements MotorTankGeometry
- implements API logger
- implements hashing for rocket model and udpates api docs adapts flight controller to hybrid
- adapt updates to hybrid
- updates flight controller to support hybrid updates start saving rocket option and motor kind
- adapts flight to save rocket options and motor kind state

### CODE REVIEW

**Review by luimot — COMMENTED** [review]
In summary it's all fine but I guess names like "Callisto" and "Cesaroni" and other default things could be on separate folders

File: lib/controllers/rocket.py [hunk 1]
```diff
@@ -28,13 +29,13 @@ class RocketController():
     Enables:
        create a RocketpyRocket object from a Rocket model object.
     """
-    def __init__(self, rocket: Rocket):
+    def __init__(self, rocket: Rocket, rocket_option, motor_kind):
```
Comment (Gui-FernandesBR) [inline lib/controllers/rocket.py:32]: What is `rocket_option` ?

Comment (Gui-FernandesBR) [reply lib/controllers/rocket.py:32]: What values can this assume?

File: lib/controllers/flight.py [hunk 2]
```diff
@@ -5,7 +5,8 @@
 
 import jsonpickle
 
-from lib.models.rocket import Rocket
+from lib.models.rocket import Rocket, RocketOptions
+from lib.models.motor import Motor, MotorEngines, MotorKinds
 from lib.models.flight import Flight
 from lib.models.environment import Env
 from lib.views.flight import FlightSummary, SurfaceWindConditions, OutOfRailConditions, BurnoutConditions, ApogeeConditions, MaximumValues, InitialConditions, NumericalIntegrationSettings, ImpactConditions, EventsRegistered, LaunchRailConditions, FlightData, FlightCreated, FlightUpdated, FlightDeleted, FlightPickle
```
Comment (Gui-FernandesBR) [inline lib/controllers/flight.py:12]: This is an example of too long line. The code would benefit from a linter like black

https://black.readthedocs.io/en/stable/

File: lib/repositories/rocket.py
[the remark this answers is not in the document]
Comment (GabrielBarberini) [reply lib/repositories/rocket.py:1]: I was thinking Calisto is a valid rocket option, isn't it? Since it is in the data files. Hence I was thinking of it more like a default value, I mean, the default value between the possible valid choices for rocket (including the future off-shelf choices),

File: lib/controllers/rocket.py (hunk 1 above)
Comment (GabrielBarberini) [reply lib/controllers/rocket.py:32]: It is the rocket to be selected from the list of rockets we have available under /data e.g: `power_off_drag=f"lib/data/{rocket_option.value.lower()}/powerOffDragCurve.csv`, for now the only available options are:

```
class RocketOptions(str, Enum):
    calisto: str = "Calisto"
    custom: str = "Custom"
```

And default is set to be "Calisto" as it is the reference rocket used in the getting started docs.

In the future, if Custom is selected, the API will expect the user to provide the powerOffDrag and powerOnDrag curves as arrays in the request.

File: lib/controllers/flight.py (hunk 2 above)
Comment (GabrielBarberini) [reply lib/controllers/flight.py:12]: Nice catch, I forgot about the imports. I think there is a flag for that in pylint as well, I will check.

File: lib/models/rocket.py
[the remark this answers is not in the document]
Comment (GabrielBarberini) [reply lib/models/rocket.py:1]: The available keys for motor_engines and rocket_options are:

motor_engines
```
class MotorEngines(str, Enum):
    cesaroni: str = "Cesaroni_M1670"
    custom: str = "Custom"
```

rocket_options
```
class RocketOptions(str, Enum):
    calisto: str = "Calisto"
    custom: str = "Custom"
```

So in fact they are both part of the class, rocket_options is an enumerable that is part of the rocket class, it is a list of accepted values for the `rocket_option` query string just like for motor_kind.

`motor_engine` is an exception in the sense that it is a list of accepted values not on the query string, but on the body, because `motor_engine` eventually translates to `thrust_source`

[the remark this answers is not in the document]
Comment (luimot) [reply lib/models/rocket.py:1]: That's alright, as discussed in meeting this will become clearer in documentation in the future, fine by me!

File: lib/secrets.py [hunk 3]
```diff
@@ -15,7 +15,6 @@ def get_os_secret(key):
     def get_secret(self, key):
         dotenv_secret = self.secrets.get(key)
         if not dotenv_secret:
-            print("cheguei")
```
Comment (luimot) [inline lib/secrets.py:18]: 😂

File: lib/controllers/flight.py
Comment (luimot) [inline lib/controllers/flight.py:1]: This is great!

File: README.md
Comment (luimot) [inline README.md:1]: Nice, readme's updated!

**Review by luimot — DISMISSED** [review]

**Review by luimot — APPROVED** [review]
